ledger-service/http-json/src/main/scala/com/digitalasset/http/PartiesService.scala

```diff
@@ -74,18 +74,27 @@ class PartiesService(
 
   private def findUnknownParties(
       found: Set[domain.PartyDetails],
-      requested: Set[domain.Party]
-  ): Set[domain.Party] =
-    if (found.size == requested.size) Set.empty[domain.Party]
-    else requested -- found.map(_.identifier)
+      requested: OneAnd[Set, domain.Party]
+  ): Set[domain.Party] = {
+    val requestedIds: Set[domain.Party] = requested.tail + requested.head
 
+    if (found.size == requestedIds.size) Set.empty
+    else requestedIds -- found.map(_.identifier)
+  }
 }
 
 object PartiesService {
   import com.digitalasset.http.util.ErrorOps._
 
   private type ET[A] = EitherT[Future, Error, A]
 
+  @SuppressWarnings(Array("org.wartremover.warts.Any"))
+  def toLedgerApi(ps: OneAnd[Set, domain.Party]): InvalidUserInput \/ OneAnd[Set, Ref.Party] = {
```

Comment:
@S11001001 Is there a better way to traverse `OneAnd[Set, X]`???

Reply:
Suggestion in another comment; you can choose any functor you want, vector, immarray, whatever, but you have the essence of the mechanics right, you have to choose something other than Set.